src/test/java/io/strimzi/kafka/bridge/http/HttpBridgeTest.java

```diff
@@ -3396,4 +3399,110 @@ void subscribeExclusiveTopicAndPattern(VertxTestContext context) throws Throwabl
+                        JsonObject bridgeResponse = response.body();
+                        String consumerInstanceId = bridgeResponse.getString("instance_id");
+                        String consumerBaseUri = bridgeResponse.getString("base_uri");
+                        assertEquals(name, consumerInstanceId);
+                        assertEquals(baseUri, consumerBaseUri);
+                    });
+
+                    create.complete(true);
+                });
+
+        create.get(TEST_TIMEOUT, TimeUnit.SECONDS);
+
+        // After
+        Thread.sleep(timeout - 1000);
+
+        //////////////////
+
+        JsonObject json = new JsonObject();
+        json.put("name", name);
+
+        // subscribe to a topic
+        JsonArray topics = new JsonArray();
+        topics.add(topic);
+
+        JsonObject topicsRoot = new JsonObject();
+        topicsRoot.put("topics", topics);
+
+
+
+        CompletableFuture<Boolean> subscribe = new CompletableFuture<>();
+        postRequest(baseUri + "/subscription")
+                .putHeader("Content-length", String.valueOf(topicsRoot.toBuffer().length()))
+                .putHeader("Content-type", BridgeContentType.KAFKA_JSON)
+                .as(BodyCodec.jsonObject())
+                .sendJsonObject(topicsRoot, ar -> {
+                    context.verify(() -> {
+                        assertTrue(ar.succeeded());
+                        assertEquals(HttpResponseStatus.NO_CONTENT.code(), ar.result().statusCode());
+                    });
+                    subscribe.complete(true);
+                });
+
+        subscribe.get(TEST_TIMEOUT, TimeUnit.SECONDS);
+
+        Thread.sleep(timeout + 2000);
+
+
+        //////////////////
+
+
+
+
+
+        CompletableFuture<Boolean> subscribeAfterTimeout = new CompletableFuture<>();
+        postRequest(baseUri + "/subscription")
+                .putHeader("Content-length", String.valueOf(topicsRoot.toBuffer().length()))
+                .putHeader("Content-Type", BridgeContentType.KAFKA_JSON)
+                .as(BodyCodec.jsonObject())
+                .sendJsonObject(topicsRoot, ar -> {
+                    context.verify(() -> {
+                        assertTrue(ar.succeeded());
+                        HttpResponse<JsonObject> response = ar.result();
+                        HttpBridgeError error = HttpBridgeError.fromJson(response.body());
+                        assertEquals(HttpResponseStatus.NOT_FOUND.code(), response.statusCode());
+                        assertEquals(HttpResponseStatus.NOT_FOUND.code(), error.getCode());
+                        assertEquals("The specified consumer instance was not found.", error.getMessage());
+                    });
+                    subscribeAfterTimeout.complete(true);
+                });
+        subscribeAfterTimeout.get(TEST_TIMEOUT, TimeUnit.SECONDS);
+        context.completeNow();
+
+        this.bridgeConfig.getConfig().remove(HttpConfig.HTTP_CONSUMER_TIMEOUT);
```

Comment:
do you need this? I don't think that the other tests don't have activity for 5 seconds.